src/pykiso/lib/auxiliaries/ykush_auxiliary.py

```diff
@@ -0,0 +1,409 @@
+            # blocking by default
+            self._ykush_device = hid.device()
+            self._ykush_device.open_path(path)
+            self._path = path
+        else:
+            # otherwise try to locate a device
+            for device in hid.enumerate(0, 0):
+                if (
+                    device["vendor_id"] == YKUSH_USB_VID
+                    and device["product_id"] in YKUSH_USB_PID_LIST
+                ):
+                    list_ykush_device.append(device["serial_number"])
+                    if serial is None or serial == device["serial_number"]:
+                        # Alber3.1 3.2 object attribute for YKUSHXS
+                        self._product_id = device["product_id"]
+                        return self.find_device(path=device["path"])
+        if self._ykush_device is None:
+            if list_ykush_device == []:
+                raise YkushDeviceNotFound(
+                    "Could not connect to a ykush hub, no device was found."
+                )
+            else:
+                raise YkushDeviceNotFound(
+                    f"The serial numbers available are : {list_ykush_device}\n"
+                    f"No device was found with the serial number {serial}\n"
+                    if serial
+                    else "",
+                )
+
+    @contextmanager
+    def _open_and_close_device(self):
+        """Context manager to open and close device every time we send a message
+        else we will get an empty message every time in response.
+        """
+        self._ykush_device = hid.device()
+        self._ykush_device.open_path(self._path)
+        try:
+            yield
+        finally:
+            self._ykush_device.close()
+
+    def check_port_number(self, port_number: int):
+        """Check if the port indicated is a port of the device
+
+        :raises YkushPortNumberError: Raise error if no port has this number
+        """
+        if port_number not in range(1, self.number_of_port + 1):
+            raise YkushPortNumberError(
+                f"The port number {port_number} is not valid for the device,"
+                f" it has only {self.number_of_port} ports"
+            )
+
+    @staticmethod
+    def get_str_state(state: int) -> str:
+        """Return the str of a state
+
+        :param state: 1 (port on), 0 (port off)
+        :return: ON, OFF
+        """
+        return YKUSH_PORT_STATE_DICT[state]
+
+    def get_serial_number_string(self) -> str:
+        """Returns the device serial number string"""
+        with self._open_and_close_device():
+            return self._ykush_device.get_serial_number_string()
+
+    def get_number_of_port(self) -> int:
+        """Returns the number of port on the ykush device"""
+        # original YKUSH 1,3 port count
+        count = 3
+        # YKUSHXS Make it have 1 port
+        if self._product_id == 0xF0CD:
+            count = 1
+
+        return count
+
+    def get_port_state(self, port_number: int) -> int:
```

Comment:
You could also group both of these methods by returning an Enum:
```
class PortState(IntEnum):
    OFF = 0
    ON = 1
```

Seems a bit boilerplate to me to always duplicate the API just to change the return value's type

Reply:
Done